frappe/model/document.py

```diff
@@ -49,6 +49,9 @@ def __value__(self):
 		# Used when requiring its value representation for db interactions, serializations, etc
 		return self.name
 
+	def __hash__(self):
+		return hash(self.doctype + self.name or "")
```

Comment:
This is another ill-thought change. All unsaved documents now map to same hash?

Why is this change even required?

Reply:
That's not good. Thanks for the spot.